MetaMask/metamask-extension
# Update changelog headers and fix dates

The changelog release header format has been updated to match the ["keep a changelog" ](https://keepachangelog.com/en/1.0.0/) format. Each header is now the bracketed version number followed by a dash, then the release date in ISO-8601 format.

The release dates in each header were also updated to match the date of the corresponding [GitHub Release](https://github.com/MetaMask/metamask-extension/releases). Many of these dates were incorrect because they were set on the day we created the release candidate, rather than on the day of release.

Any changelog release entries without a corresponding GitHub release was left with the date already specified.

The three oldest release headers were missing dates. For the first two, I used the date of the version bump commit. For the third, I removed it since no changes were listed anyway, and it represented a range of releases rather than a single one.

The `auto-changelog.js` script has been updated to account for this new format as well.

Manual testing steps: 

For RC changelog updates with existing release header:
- Bump the `version` in `app/manifest/_base.json`, to simulate an RC-like environment
- Add a new release header to the changelog for the new version
- Run `yarn update-changelog`
- See that it places changelog entries under the new release header
 
For updates without a release header:
- Bump the `version` in `app/manifest/_base.json`, to simulate an RC-like environment
- Run `yarn update-changelog`
- See that it places changelog entries under the newly added release header
